extensions/rich_text_components/Video/Video.js

```diff
@@ -25,16 +25,44 @@ oppia.directive('oppiaNoninteractiveVideo', [
       restrict: 'E',
       scope: {},
       templateUrl: 'richTextComponent/Video',
-      controller: ['$scope', '$attrs', 'explorationContextService',
-        function($scope, $attrs, explorationContextService) {
+      controller: ['$scope', '$attrs', 'explorationContextService', '$element',
+        'PAGE_CONTEXT',
+        function($scope, $attrs, explorationContextService, $element) {
           var start = oppiaHtmlEscaper.escapedJsonToObj($attrs.startWithValue);
           var end = oppiaHtmlEscaper.escapedJsonToObj($attrs.endWithValue);
 
-          $scope.autoplaySuffix = (oppiaHtmlEscaper.escapedJsonToObj(
-            $attrs.autoplayWithValue) ? '&autoplay=1' : '&autoplay=0');
           $scope.videoId = oppiaHtmlEscaper.escapedJsonToObj(
             $attrs.videoIdWithValue);
           $scope.timingParams = '&start=' + start + '&end=' + end;
+
+          // This code helps in checking whether video is for upcoming content.
+          // This is totally dependent of parent div of video container
+          // If in future that parent div changes then this code will be
+          // required to change accordingly.
+          var isVisible = true;
+          var videoVisibilty = angular.element($element).prop('offsetParent');
```

Comment:
Thanks for checking! I think it's fine to not worry too much about the youtube/html5 video player, for now -- let's deal with that another time. It actually seems like a bit of a bug to me -- when youtube is given an "autoplay" attribute, it should probably autoplay, regardless of whether the flash or html5 player is being used.

Btw, the proposal I made was also meant to apply to the case of small desktop screens, e.g. if there's a bunch of content and the video is "below the fold".